toolchain/check/eval_inst.cpp

```diff
@@ -203,21 +211,34 @@ auto EvalConstantInst(Context& context, SemIR::FacetAccessType inst)
   return ConstantEvalResult::NewSamePhase(inst);
 }
 
+auto EvalConstantInst(Context& context, SemIR::FacetValue inst)
+    -> ConstantEvalResult {
+  // A FacetValue that just wraps a BindSymbolicName without adding/removing any
+  // witnesses is evaluated back to the BindSymbolicName itself.
+  if (auto access =
+          context.insts().TryGetAs<SemIR::FacetAccessType>(inst.type_inst_id)) {
+    auto bind_id = access->facet_value_inst_id;
+    auto bind = context.insts().Get(bind_id);
+    if (bind.Is<SemIR::BindSymbolicName>()) {
+      // If the FacetTypes are the same, then the FacetValue didn't add/remove
+      // any witnesses.
+      if (bind.type_id() == inst.type_id) {
```

Comment:
Maybe bundle this together and reduce nesting, such as:

```suggestion
    // If the FacetTypes are the same, then the FacetValue didn't add/remove
    // any witnesses.
    if (auto bind = context.insts().TryGetAs<SemIR::BindSymbolicName>(bind_id); bind && bind.type_id() == inst.type_id) {
```

Reply:
The reduced nesting is nice but I am kind of averse to using the `if (.. ; ..)` syntax just cuz it's so easy to forget to test the thing on the lhs of the `;` which is automatically tested without the `;`. So idk if it's better overall..

Reply:
You could also move the `auto bind` out of the if -- mainly I'm suggesting the `TryGetAs`, and use of `&&` instead of nested `if`

Reply:
Done